indigo_app/forms/works.py

```diff
@@ -576,11 +575,22 @@ def clean_all_provisions(self):
         return self.cleaned_data['all_provisions']
 
     def clean(self):
-        super().clean()
+        cleaned_data = super().clean()
         # all_provisions may have been nuked during clean
-        if self.cleaned_data.get('all_provisions') and self.cleaned_data['provisions']:
+        if cleaned_data.get('all_provisions') and cleaned_data['provisions']:
             raise ValidationError(_("Cannot specify all provisions, and a list of provisions."))
 
+        # don't try to save a duplicate commencement, we'll get an IntegrityError
+        duplicate_commencement = self.work.commencements.exclude(
+            pk=self.instance.pk
+        ).filter(
+            commencing_work=cleaned_data.get('commencing_work'),
+            date=cleaned_data.get('date')).first()
```

Comment:
minor: use `.exists()` when you want to check if it exists without actually fetching any data.